translation/translator.go

```diff
@@ -81,38 +81,60 @@ func (t *Translator) Get(key string, options ...translationcontract.Option) (str
 	// simply load the file and return the line if it exists.
 	// If the file doesn't exist, we will return fallback if it is enabled.
 	// Otherwise, we will return the key as the line.
-	if err := t.load(folder, locale); err != nil && err != ErrFileNotExist {
-		return "", err
-	}
-
-	line := t.loaded[folder][locale][keyPart]
-	if line == "" {
-		fallbackFolder, fallbackLocale := parseKey(t.GetFallback())
-		// If the fallback locale is different from the current locale, we will
-		// load in the lines for the fallback locale and try to retrieve the
-		// translation for the given key.If it is translated, we will return it.
-		// Otherwise, we can finally return the key as that will be the final
-		// fallback.
-		if (folder+locale != fallbackFolder+fallbackLocale) && fallback {
-			var fallbackOptions translationcontract.Option
-			if len(options) > 0 {
-				fallbackOptions = options[0]
+	if err := t.load(locale, folder); err != nil && err != ErrFileNotExist {
+		//facades.Log().Error(err)
+		return t.key
```

Comment:
Q: Why return directly? Should call fallback.

Reply:
We will callback only if err is of type `ErrFileNotExist` else there might be some other error when loading file. should we call fallback for every error?

Reply:
Yes, we need to call fallback for every error.

Reply:
Actually when we are returning key our actual intention is to let user know that there is something error in your file. If we are returning fallback there might be case that fallback is present and user miss the error. So i think this will defeat the actual purpose of returning key.

Reply:
Sorry, based on the Laravel code, I think it's better to panic when encountering an invalid JSON error.

Reply:
okay